Jenkinsfile

```diff
@@ -52,6 +52,31 @@ pipeline {
       }
     }
 
+    stage('Publish on ECR') {
+      agent { label 'sbt-template' }
+      environment {
+        NEXUS = "${env.NEXUS}"
+        NEXUS_CREDENTIALS = credentials('pdnd-nexus')
+        DOCKER_REPO = "${env.DOCKER_REPO}"
+        MAVEN_REPO = "${env.MAVEN_REPO}"
+        ECR_RW = credentials('ecr-rw')
+        PDND_TRUST_STORE_PSW = credentials('pdnd-interop-trust-psw')
+      }
+      steps {
+        container('sbt-container') {
```

Comment:
could we be more idiomatic here? I mean using `withCredentials` binding instead of manually exporting variables in sh script? I mean something like:

```
withCredentials([usernamePassword(credentialsId: 'ecr-rw', usernameVariable: 'AWS_ACCESS_KEY_ID', passwordVariable: 'AWS_SECRET_ACCESS_KEY')]) {
  // now variable name and password name should be bound to sh step.
}
```

Reply:
Nice 👍